fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryInsertPlan.java

```diff
@@ -0,0 +1,161 @@
+        return store.saveRecordAsync(message, FDBRecordStoreBase.RecordExistenceCheck.ERROR_IF_EXISTS);
+    }
+
+    @Nonnull
+    @Override
+    public RecordQueryInsertPlan translateCorrelations(@Nonnull final TranslationMap translationMap,
+                                                       @Nonnull final List<? extends Quantifier> translatedQuantifiers) {
+        return new RecordQueryInsertPlan(
+                Iterables.getOnlyElement(translatedQuantifiers).narrow(Quantifier.Physical.class),
+                getTargetRecordType(),
+                getTargetType(),
+                getTargetDescriptor(),
+                getCoercionTrie(),
+                getComputationValue());
+    }
+
+    @Nonnull
+    @Override
+    public RecordQueryInsertPlan withChild(@Nonnull final RecordQueryPlan child) {
+        return new RecordQueryInsertPlan(Quantifier.physical(GroupExpressionRef.of(child)),
+                getTargetRecordType(),
+                getTargetType(),
+                getTargetDescriptor(),
+                getCoercionTrie(),
+                getComputationValue());
+    }
+
+    @Override
+    public int hashCodeWithoutChildren() {
+        return Objects.hash(BASE_HASH.planHash(), super.hashCodeWithoutChildren());
+    }
+
+    @Override
+    public int planHash(@Nonnull final PlanHashKind hashKind) {
+        return PlanHashable.objectsPlanHash(hashKind, BASE_HASH, super.planHash(hashKind));
+    }
+
+    @Nonnull
+    @Override
+    public String toString() {
+        return getInnerPlan() + " | " + "INSERT INTO " + getTargetRecordType();
+    }
+
+    /**
+     * Rewrite the planner graph for better visualization.
+     * @param childGraphs planner graphs of children expression that already have been computed
+     * @return the rewritten planner graph that models the filter as a node that uses the expression attribute
+     *         to depict the record types this operator filters.
+     */
+    @Nonnull
+    @Override
+    public PlannerGraph rewritePlannerGraph(@Nonnull List<? extends PlannerGraph> childGraphs) {
+        final var graphForTarget =
+                PlannerGraph.fromNodeAndChildGraphs(
+                        new PlannerGraph.DataNodeWithInfo(NodeInfo.BASE_DATA,
+                                getResultType(),
+                                ImmutableList.of(getTargetRecordType())),
+                        ImmutableList.of());
+
+        return PlannerGraph.fromNodeInnerAndTargetForModifications(
+                new PlannerGraph.ModificationOperatorNodeWithInfo(this,
+                        NodeInfo.MODIFICATION_OPERATOR,
+                        ImmutableList.of("INSERT"),
+                        ImmutableMap.of()),
+                Iterables.getOnlyElement(childGraphs), graphForTarget);
+    }
+
+    /**
+     * Factory method to create a {@link RecordQueryInsertPlan}.
```

Comment:
I'm not opposed, but other plan classes don't follow this pattern. Perhaps they should?